### PR TITLE
added model, resources and tests

### PR DESCRIPTION
**Pull request by vdprogr**

### CODE REVIEW

File: src/main/java/SimpleApplication.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    public static void main( String[] args )
+    {
+        try {
+            new SimpleApplication().run(args);
+        }
+        catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
```
Comment (romanista) [inline src/main/java/SimpleApplication.java]: Please make it more readable. Like this
```
    public static void main(String[] args) throws Exception {
        new SimpleApplication().run(args);
    }
```

File: src/main/java/SimpleConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+
```
Comment (romanista) [inline src/main/java/SimpleConfiguration.java]: Move file to config/configuration package

- [x] Move to config/configuration package

File: src/main/java/com/dropwizard/example/h2/DAO/PersonDAO.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package com.dropwizard.example.h2.DAO;
```
Comment (romanista) [inline src/main/java/com/dropwizard/example/h2/DAO/PersonDAO.java]: Don't name packages with upper cases. As well as classes. 

> DAO -> dao
> PersonDAO -> PersonDao

- [x] Rename package and class

File: src/main/java/com/dropwizard/example/h2/DAO/PersonDAO.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import java.util.List;
+import java.util.Optional;
+
+public interface PersonDAO {
+
+    @SqlQuery("select * from person where id = :id")
+    @RegisterRowMapper(PersonMapper.class)
+    Optional<Person> getById(@Bind("id") Long id);
+
+    @SqlQuery("select id, firstname, lastname from person")
```
Comment (romanista) [inline src/main/java/com/dropwizard/example/h2/DAO/PersonDAO.java]: In line 17 you have wild (*) select and in Line 21 you listed all columns explicitly. Any reason behind that?

File: src/main/java/com/dropwizard/example/h2/model/Person.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
+package com.dropwizard.example.h2.model;
```
Comment (romanista) [inline src/main/java/com/dropwizard/example/h2/model/Person.java]: Let's write Builder for class Person! And add unit tests for it.

- [ ] Add Builder

File: src/main/java/com/dropwizard/example/h2/resources/PersonResource.java [hunk 6]
```diff
@@ -0,0 +1,97 @@
+package com.dropwizard.example.h2.resources;
```
Comment (romanista) [inline src/main/java/com/dropwizard/example/h2/resources/PersonResource.java]: Person and PersonResource are not part of h2 at all, thus please place it in proper package. Like
> com.dropwizard.example.model
> com.dropwizard.example.resource

- [x] Move to proper packages

File: src/main/java/SimpleApplication.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
+import com.dropwizard.example.h2.DAO.PersonDAO;
```
Comment (romanista) [inline src/main/java/SimpleApplication.java]: Place this class under some package. Like

> com.dropwizard.example

**Comment by romanista** [comment]
General comment. Think about packaging. I didn't put comments for all classes, but think what relates to h2 put it there, but only that and nothing else. Image in the future you will have let's say PostgreSQL db, where all needed classes will be placed then to not mix up everything? Think how to structure and organize the code right now.

File: src/main/java/com/dropwizard/example/h2/resources/PersonResource.java [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+        return Response.ok(persons).build();
+    }
+
+    @GET
+    @Timed
+    @Path("{id}")
+    public Response getPersonById(@PathParam("id") Long id) {
+
+        LOGGER.info("Fetching person by id {} ", id);
+        Optional<Person> personFromDb = personDAO.getById(id);
```
Comment (romanista) [inline src/main/java/com/dropwizard/example/h2/resources/PersonResource.java]: No need to name variable as _personFromDb_, just _person_ is enough.

File: src/main/java/com/dropwizard/example/h2/resources/PersonResource.java [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+        if (personFromDb.isPresent()) {
```
Comment (romanista) [inline src/main/java/com/dropwizard/example/h2/resources/PersonResource.java:45]: Try to rewrite it more in functional style.

### **Hint**. orElseGet()

- [ ] Rewrite in functional style

File: src/main/java/com/dropwizard/example/h2/resources/PersonResource.java [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+            return Response.ok(personFromDb).build();
+        }
+        else {
+
+            return Response
+                    .status(Response.Status.NOT_FOUND)
+                    .entity("Person with ID = "+id+" not found !")
+                    .build();
+        }
+    }
```
Comment (romanista) [inline src/main/java/com/dropwizard/example/h2/resources/PersonResource.java]: Try to rewrite it more in functional style.

_Hint_. orElseGet()

- [ ] Rewrite in functional style

File: src/main/java/com/dropwizard/example/h2/resources/PersonResource.java [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+        if (personFromDb.isPresent()) {
+            return Response.ok(personFromDb).build();
+        }
+        else {
+            return Response
+                    .status(Response.Status.NOT_FOUND)
+                    .entity("Person with ID = "+id+" not found for update !")
+                    .build();
+        }
+    }
```
Comment (romanista) [inline src/main/java/com/dropwizard/example/h2/resources/PersonResource.java]: Please check this [comment](https://github.com/vdprogr/Dropwizard-rest-H2/pull/1/files#r796484980)

- [ ] Reimplement with functional style.

File: src/test/java/com/dropwizard/example/h2/model/PersonTest.java [hunk 12]
```diff
@@ -0,0 +1,62 @@
+package com.dropwizard.example.h2.model;
```
Comment (romanista) [inline src/test/java/com/dropwizard/example/h2/model/PersonTest.java]: Use JUnit5.

- [x] Add dependency for JUnit5

File: src/test/java/com/dropwizard/example/h2/model/PersonTest.java [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+
+    private Person person;
+
+    @Before
+    public void setUp(){
+        person = new Person();
+    }
+
+    @Test
+    public void test_constructor(){
```
Comment (romanista) [inline src/test/java/com/dropwizard/example/h2/model/PersonTest.java]: Tests should be named like this _testConstructor_. No underscores

- [x] Rename test methods.

File: src/test/java/com/dropwizard/example/h2/model/PersonTest.java [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+        final String lastname = "Test lastname";
+
+        final Person myPerson = new Person(id, firstname, lastname);
+
+        assertEquals(id, myPerson.getId());
+        assertEquals( firstname, myPerson.getFirstName());
+        assertEquals( lastname, myPerson.getLastName());
+
+    }
+
```
Comment (romanista) [inline src/test/java/com/dropwizard/example/h2/model/PersonTest.java]: Please add test for 

> new Person(null, null, null)

- [ ] Add test for new Person(null, null, null)

File: src/test/java/com/dropwizard/example/h2/resources/PersonResourceTest.java [hunk 15]
```diff
@@ -0,0 +1,150 @@
+package com.dropwizard.example.h2.resources;
```
Comment (romanista) [inline src/test/java/com/dropwizard/example/h2/resources/PersonResourceTest.java]: Ok, here you already have JUnit5)) Then clean up dependencies and just fix PersonTest.

File: src/test/java/com/dropwizard/example/h2/resources/PersonResourceTest.java [hunk 16]
```diff
@@ -0,0 +1,150 @@
...
+
+    }
+
+    @AfterEach
+    void tearDown() {
+        reset(personDAO);
+    }
+
+    @Test
+    public void getAllWithTwoElements() {
```
Comment (romanista) [inline src/test/java/com/dropwizard/example/h2/resources/PersonResourceTest.java]: What is this test name about and same for below one test?

File: src/test/java/com/dropwizard/example/h2/resources/PersonResourceTest.java [hunk 17]
```diff
@@ -0,0 +1,150 @@
...
+        List<Person> persons = List.of(person1, person2);
+        when(personDAO.getAll()).thenReturn(persons);
+
+        List<Person> result = RULE.target("/persons").request().get(new GenericType<List<Person>>() {});
+
+        assertEquals(2, result.size());
+        assertEquals("firstname1", result.get(0).getFirstName());
```
Comment (romanista) [inline src/test/java/com/dropwizard/example/h2/resources/PersonResourceTest.java]: Not to duplicate strings I'd suggest to have it like this

> assertEquals(person1.getFirstName(), result.get(0).getFirstName());

File: src/test/java/com/dropwizard/example/h2/resources/PersonResourceTest.java [hunk 18]
```diff
@@ -0,0 +1,150 @@
...
+
+    }
+
+    @Test
+    public void getAllWithoutElements() {
+        List<Person> persons = List.of();
+        when(personDAO.getAll()).thenReturn(persons);
+
+        List<Person> result = RULE.target("/persons").request().get(new GenericType<List<Person>>() {});
+        assertEquals(0, result.size());
```
Comment (romanista) [inline src/test/java/com/dropwizard/example/h2/resources/PersonResourceTest.java]: You can write it like

> assertTrue(result.isEmpty());

File: src/test/java/com/dropwizard/example/h2/resources/PersonResourceTest.java [hunk 19]
```diff
@@ -0,0 +1,150 @@
...
+        verify(personDAO, times(1)).getAll();
+    }
+
+
+    @Test
+    void getPersonSuccess() {
```
Comment (romanista) [inline src/test/java/com/dropwizard/example/h2/resources/PersonResourceTest.java]: About naming. If you test method _getPersonById_ here then name test method as _testGetPersonById_. If you want to specify result of invocation you can have it like _testGetPersonById_Success_. The rule is -> test<MethodName>_<Result or Very useful info to understand test>

**Comment by romanista** [comment]
Also as far asI can see there is no service layer. Please add it between resource and dao.

- [ ] Add service layer

**Review by romanista — CHANGES_REQUESTED** [review]
Requested changes

File: src/main/java/com/dropwizard/example/h2/dao/PersonDao.java [hunk 20]
```diff
@@ -0,0 +1,32 @@
+package com.dropwizard.example.h2.dao;
```
Comment (romanista) [inline src/main/java/com/dropwizard/example/h2/dao/PersonDao.java]: Does PersonDao relates to h2? Same for PersonMapper.

File: src/main/java/config/configuration/SimpleConfiguration.java [hunk 21]
```diff
@@ -0,0 +1,30 @@
+package config.configuration;
```
Comment (romanista) [inline src/main/java/config/configuration/SimpleConfiguration.java]: I meant just 'config' or 'configuration' package) but ok)

File: src/test/java/com/dropwizard/example/resources/PersonResourceTest.java [hunk 22]
```diff
@@ -0,0 +1,146 @@
...
+        List<Person> result = RULE.target("/persons").request().get(new GenericType<List<Person>>() {});
+
+        assertEquals(2, result.size());
+        assertEquals(person1.getFirstName(), result.get(0).getFirstName());
+        assertEquals(person2.getFirstName(), result.get(1).getFirstName());
+    }
+
+    @Test
+    public void testGetAllPersonsIfEmpty() {
+        List<Person> persons = List.of();
```
Comment (romanista) [inline src/test/java/com/dropwizard/example/resources/PersonResourceTest.java]: List.of() better to replace with Collections.emptyList();

File: src/test/java/com/dropwizard/example/integration/IntegrationTest.java [hunk 23]
```diff
@@ -0,0 +1,148 @@
...
+
+    @AfterAll
+    public static void tearDown() throws IOException {
+        client.close();
+    }
+
+    @Test
+    void testGetPersonSuccess() {
+        // try to get person with ID=1
+        final Person person = new Person(1L, "Dan", "Johnson");
```
Comment (romanista) [inline src/test/java/com/dropwizard/example/integration/IntegrationTest.java:46]: Why you need create this person object? Just declare it's properties (id, first name, last name). Below are some more tests with this

File: src/test/java/com/dropwizard/example/integration/IntegrationTest.java [hunk 24]
```diff
@@ -0,0 +1,148 @@
...
+
+        final Person newPerson = client.target("http://localhost:" + EXT.getLocalPort() + "/persons/1")
```
Comment (romanista) [inline src/test/java/com/dropwizard/example/integration/IntegrationTest.java]: Extract "http://localhost" to const

File: src/test/java/com/dropwizard/example/integration/IntegrationTest.java [hunk 25]
```diff
@@ -0,0 +1,148 @@
...
+                .request()
+                .get();
+
+        assertEquals(Response.Status.NOT_FOUND.getStatusCode(), response.getStatusInfo().getStatusCode());
+    }
+
+    @Test
+    void testGetAllPerson() {
+
+        final Response response =
```
Comment (romanista) [inline src/test/java/com/dropwizard/example/integration/IntegrationTest.java:74]: And how many persons there?

File: src/test/java/com/dropwizard/example/integration/IntegrationTest.java [hunk 26]
```diff
@@ -0,0 +1,148 @@
...
+                .request()
+                .put(Entity.entity(person, MediaType.APPLICATION_JSON_TYPE));
+
+        assertEquals(Response.Status.NOT_FOUND.getStatusCode(), response.getStatusInfo().getStatusCode());
+    }
+
+
+    @Test
+    void testDeletePersonSuccess() {
+        // try to delete person with ID=2
```
Comment (romanista) [inline src/test/java/com/dropwizard/example/integration/IntegrationTest.java:128]: You can check also how many where before deletion and how many left after
